File: examples/nodejs/redis-queue/worker.js
import Redis from 'ioredis';

const redis = new Redis({ host: process.env.REDIS_HOST || 'localhost', port: 6379, password: process.env.REDIS_PASSWORD || 'change_me' });
const queue = process.env.REDIS_QUEUE || 'jobs:email';

console.log('[worker] waiting job...');
const data = await redis.brpop(queue, 0);
console.log('[worker] got:', data?.[1]);
await redis.quit();

File: examples/python/redis-queue/worker.py
import os
import redis

r = redis.Redis(
    host=os.getenv("REDIS_HOST", "localhost"),
    port=6379,
    password=os.getenv("REDIS_PASSWORD", "change_me"),
    decode_responses=True,
)

print("[worker] waiting...")
item = r.brpop("jobs:email", timeout=0)
print("[worker] got", item)
